sympy/series/formal.py

```diff
@@ -0,0 +1,1107 @@
+        as a polynomial(without ``O`` term).
+        """
+        terms = []
+        for i, t in enumerate(self):
+            xp = self._get_pow_x(t)
+            if xp >= n:
+                break
+            elif xp.is_integer is True and i == n + 1:
+                break
+            elif t is not S.Zero:
+                terms.append(t * self.mul)
+
+        return Add(*terms)
+
+    def truncate(self, n=6):
+        """Truncated series.
+
+        Returns truncated series expansion of f upto
+        order ``O(x**n)``.
+
+        If n is ``None``, returns an infinite iterator.
+        """
+        if n is None:
+            return iter(self)
+
+        x, x0 = self.x, self.x0
+        pt_xk = self.xk.coeff(n)
+        if x0 is S.NegativeInfinity:
+            x0 = S.Infinity
+
+        return self.polynomial(n) + self.mul * Order(pt_xk, (x, x0))
+
+    def _eval_term(self, pt):
+        pt_xk = self.xk.coeff(pt)
+
+        try:
+            pt_ak = self.ak.coeff(pt).simplify()  # TODO: Thoughts?
+        except IndexError:
+            pt_ak = S.Zero
+
+        term = (pt_ak * pt_xk)
+
+        if self.ind:
+            ind = S.Zero
+            for t in Add.make_args(self.ind):
+                pow_x = self._get_pow_x(t)
+                if pt == 0 and pow_x < 1:
+                    ind += t
+                elif pow_x >= pt and pow_x < pt + 1:
+                    ind += t
+            term += ind
+
+        return term.collect(self.x)
+
+    def _eval_subs(self, old, new):
+        x = self.x
+        if old.has(x):
+            return self
+
+    def _eval_as_leading_term(self, x):
+        for t in self:
+            if t is not S.Zero:
+                return t
+
+
+def fps(f, x=None, x0=0, dir=1, hyper=True, order=4, rational=True, full=False):
+    """Generates Formal Power Series of f.
+
+    Returns the formal series expansion of ``f`` around ``x = x0``
+    with respect to ``x`` in the form of a ``FormalPowerSeries`` object.
+
+    Formal Power Series is represented using an explicit formula
+    computed using different algorithms.
+
+    See :func:`compute_fps` for the more details regarding the computation
+    of formula.
+
+    Parameters
+    ==========
+
+    * x : if ``x=None`` and ``f`` is univariate, the univariate
+    symbols will be supplied, otherwise an error will be raised.
+
+    * x0 : series expansion around ``x = x0``(Default = 0).
+
+    * dir : For ``dir=1`` (default) the series is calculated from the right and
+    for ``dir=-1`` the series from the left. For smooth functions this
+    flag will not alter the results.
+
+    * hyper : set hyper=False, for not using hypergeometric algorithm.
+
+    * order : Order of the derivative of f, till which algorithms
+    are run.
+
+    * rational : set rational=False to skip rational algorithm
+
+    * full : full by default is False. Try setting full to True
+    to increase the range of rational algorithm. See :func:`rational_algorithm`
+    for details.
```

Comment:
Fix the parameters section as per https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt